statstracker/src/tests/App.test.js

```diff
@@ -0,0 +1,101 @@
+// Required to get <App /> running:
+import React from 'react';
+import { Provider } from 'react-redux';
+import { createStore } from 'redux';
+import rootReducer from '../reducers/reducers';
+import App from '../components/App';
+const store = createStore(rootReducer);
+
+import { render, fireEvent } from 'react-testing-library';
+import 'jest-dom/extend-expect';
+
+const packagedApp = (
+  <Provider store={store}>
+    <App />
+  </Provider>
+);
+
+describe('<App />', () => {
+  it('renders successfully', () => {
+    render(packagedApp);
+  });
+});
+
+describe('Strike Button', () => {
+  it('adds a strike 0 to 1', () => {
+    const { getByText } = render(packagedApp);
+    const button = getByText(/strike$/i);
+    fireEvent.click(button);
+    getByText(/strikes: 1/i);
```

Comment:
refactored the regex to make sure it only matches the button  -- we've also seen that you can pass in the exact text as a string to match as well, but that will be more brittle in the long term in case even a small change happens in the app, like casing changing

Reply:
in my opinion, the best way to test UI like this is to verify the default state exists (check that it's first "Strike: 0"), then simulate a button click, then test for what you expect the text to now be